### PR TITLE
🐛 [RUM-3599] do not define undefined instrumented method

### PR DESCRIPTION
**Pull request by BenoitZugmeyer**
## Motivation

Some websites are removing native methods on purpose. Re-defining the method breaks them.

## Changes

Don't define undefined instrumented methods.

## Testing

- [x] Local
- [ ] Staging
- [x] Unit
- [ ] End to end

---

I have gone over the [contributing](https://github.com/DataDog/browser-sdk/blob/main/CONTRIBUTING.md) documentation.

### CODE REVIEW

**Review by N-Boutaib — APPROVED** [review]

File: packages/core/src/tools/instrumentMethod.spec.ts [hunk 1]
```diff
@@ -28,15 +28,23 @@ describe('instrumentMethod', () => {
...
-  it('sets a method originally undefined', () => {
+  it('does not set a method originally undefined', () => {
     const object: { method?: () => number } = {}
 
+    instrumentMethod(object, 'method', noop)
+
+    expect(object.method).toBeUndefined()
+  })
+
+  it('sets an event handler even if it was originally undefined', () => {
```
Comment (thomas-lebeau) [inline packages/core/src/tools/instrumentMethod.spec.ts:39]: **❓ question:** ‏ could you explain when this happens? and why we still want to wrap something that was originally undefined?

**Review by thomas-lebeau — APPROVED** [review]

**Pull request merged**